lib/Exporter/ONNXModelWriter.cpp

```diff
@@ -108,6 +158,28 @@ void outputsToProto(const Node *node, ONNX_NAMESPACE::NodeProto *proto) {
   }
 }
 
+/// Write the output of the provided type only.
+bool outputKindToProto(Kinded::Kind kind, const Node *node,
+                       ONNX_NAMESPACE::NodeProto *proto) {
+  bool found = false;
+  for (const auto &use : node->getUsers()) {
+    const auto *user = use.getUser();
+    if (user->getKind() == Kinded::Kind::SaveNodeKind) {
+      proto->add_output(user->getName());
+    } else if (user->getKind() == kind) {
+      // Find the user input that matches input node parameter.
+      for (unsigned b = 0, e = user->getNumInputs(); b < e; ++b) {
+        auto *UIN = user->getNthInput(b).getNode();
+        if (UIN == node) {
```

Comment:
I need the name of user's particular input that connects to the UIN. User can have more than one input.

Reply:
Detailed logic:
1. Loop through all users of the Parent Node.
  `for (const auto &use : node->getUsers())`
2. For each user, loop through its inputs.
  `for (unsigned b = 0, e = user->getNumInputs(); b < e; ++b)`
3. Check if the current input node is identical to the Parent Node. It can be input from other nodes.
  `if (UIN == node)`
4. Now once we found the "correct" input, get its name.
  `UIN->getName()`

Reply:
Ah, I have seen the second function, fixing it.

Reply:
Thanks for explaining this @putivsky. So my confusion is inside `outputKindToProto()`. It's guaranteed that `if (UIN == node)` will be true least once, because we already know `user` uses `node`, correct? And it seems like we don't care if we've found the "correct" input. So what is wrong with doing:

```cpp
  for (const auto &use : node->getUsers()) {
    const auto *user = use.getUser();
    if (user->getKind() == Kinded::Kind::SaveNodeKind) {
      proto->add_output(user->getName());
    } else if (user->getKind() == kind) {
      found = true;
      outputsToProto(user, proto);
    }
  }
```